app/models/amusement_park.rb

```diff
@@ -1,3 +1,13 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanics, through: :rides
+
+  def uniq_mechanics
+    mechanics.distinct
+  end
+
+  def order_rides_mechanic_experience
+    rides.joins(:mechanics).order("rides.mechanic_avg_exp")
```

Comment:
This is off to a good start with joins on `:mechanics`. The next thing that you would probably want to do is `select` something and run an `AVG` in there. Then `group`, then `order`.  It could look something like: (note this is not the only solution)
```ruby
def order_rides_mechanic_experience
  rides.joins(:mechanics)
       .select("rides.name, AVG(mechanics.years_experience) AS avg_exp")
       .group("rides.id")
       .order("avg_exp DESC")
end